maestro-ios-driver/src/main/kotlin/util/LocalSimulatorUtils.kt

```diff
@@ -171,30 +177,109 @@ object LocalSimulatorUtils {
+        MaestroTimer.withTimeout(10000) {
+            while (true) {
+                if (isAppRunning(deviceId, bundleId)) {
+                    return@withTimeout
+                } else {
+                    Thread.sleep(1000)
+                }
+            }
+        } ?: throw SimctlError("App $bundleId did not start in time")
+    }
+
+    private fun copyDirectoryRecursively(source: Path, target: Path) {
+        Files.walk(source).forEach { path ->
+            val targetPath = target.resolve(source.relativize(path).toString())
+            if (Files.isDirectory(path)) {
+                Files.createDirectories(targetPath)
+            } else {
+                Files.copy(path, targetPath)
+            }
+        }
+    }
+
+    private fun deleteFolderRecursively(folder: File): Boolean {
+        if (folder.isDirectory) {
+            folder.listFiles()?.forEach { child ->
+                deleteFolderRecursively(child)
+            }
+        }
+        return folder.delete()
+    }
+
+    private fun reinstallApp(deviceId: String, bundleId: String) {
+        val pathToBinary = Path(getAppBinaryDirectory(deviceId, bundleId))
+
+        if (Files.isDirectory(pathToBinary)) {
+            val tmpDir = createTempDirectory()
+            val tmpBundlePath = tmpDir.resolve("$bundleId-${System.currentTimeMillis()}.app")
+
+            logger.info("Copying app binary from $pathToBinary to $tmpBundlePath")
+            Files.copy(pathToBinary, tmpBundlePath)
+            copyDirectoryRecursively(pathToBinary, tmpBundlePath)
+
+            logger.info("Reinstalling and launching $bundleId")
+            uninstall(deviceId, bundleId)
+            install(deviceId, tmpBundlePath)
+            deleteFolderRecursively(tmpBundlePath.toFile())
+            logger.info("App $bundleId reinstalled and launched")
+        } else {
+            throw SimctlError("Could not find app binary for bundle $bundleId at $pathToBinary")
+        }
+    }
+
     fun clearAppState(deviceId: String, bundleId: String) {
+        logger.info("Clearing app $bundleId state")
         // Stop the app before clearing the file system
         // This prevents the app from saving its state after it has been cleared
         terminate(deviceId, bundleId)
+        ensureStopped(deviceId, bundleId)
 
-        // Wait for the app to be stopped
-        Thread.sleep(1500)
-
-        // deletes app data, including container folder
-        val appDataDirectory = getApplicationDataDirectory(deviceId, bundleId)
-        ProcessBuilder(listOf("rm", "-rf", appDataDirectory)).start().waitFor()
-
-        // forces app container folder to be re-created
-        val paths = listOf(
-            "Documents",
-            "Library",
-            "Library/Caches",
-            "Library/Preferences",
-            "SystemData",
-            "tmp"
-        )
+        // reinstall the app as that is the most stable way to clear state
+        reinstallApp(deviceId, bundleId)
+    }
```

Comment:
This could break the current maestro flows for all the users right? How do we plan to roll this out? 

Do we wait to bring on cloud/robin or we do it right away?

Should this be in next release or we include in this current version of CLI as 1.39.1?

Reply:
I am torn - the fact is that clearState is not working properly today, so the fact that we fix it means that flows that implicitly rely on the broken state of clearState might stop working. However, it's a bug fix and not a breaking change so doing a major version bump doesn't feel right. Thoughts?

Reply:
About versioning, to me it's a bug fix. Clear state being flaky since forever **is a bug**

Reply:
> Wait, how? Like, do you have any idea? We could potentially test this more to reduce this risk

People would have made their flows adapting to the current bug. So, the flows could break when this is actually fixed.


> I am torn - the fact is that clearState is not working properly today, so the fact that we fix it means that flows that implicitly rely on the broken state of clearState might stop working. However, it's a bug fix and not a breaking change so doing a major version bump doesn't feel right. Thoughts?

TBH, I am in favour of fully rolling this out without any hacks on backend. Given we emphasize that it could potentially break flows, I think its ok. 

Around the rollout, I feel this not requires a major version bump but we should not maybe just release (1.39.1) instead of waiting this to get in 1.40.0. Get this in cloud, robin and CLI as a quick follow up instead of waiting for 1.40.0 WDYT?